packages/legacy/core/App/screens/Connection.tsx

```diff
@@ -121,54 +121,78 @@ const Connection: React.FC<ConnectionProps> = ({ navigation, route }) => {
   }, [state.shouldShowDelayMessage])
 
   useEffect(() => {
-    if (!state.notificationRecord || !state.isVisible) {
+    if (!state.isVisible) {
       return
     }
 
-    // connectionless proof request, we don't have connectionless offers.
+    // If we have a connection, but no goal code, we should navigate
+    // to Chat
+    if (connection && !(Object.values(GoalCodes) as [string]).includes(oobRecord?.outOfBandInvitation.goalCode ?? '')) {
+      logger?.info('Connection: Handling connection without goal code, navigate to Chat')
+
+      dispatch({ isVisible: false })
+      navigation.getParent()?.dispatch(
+        CommonActions.reset({
+          index: 1,
+          routes: [{ name: Stacks.TabStack }, { name: Screens.Chat, params: { connectionId: connection.id } }],
+        })
+      )
+
+      return
+    }
+
+    // At this point we should be waiting for a notification
+    // to be processed
+    if (!state.notificationRecord) {
+      return
+    }
+
+    // Connectionless proof request, we don't have connectionless offers.
     if (!connection) {
       dispatch({ isVisible: false })
       navigation.replace(Screens.ProofRequest, { proofId: state.notificationRecord.id })
```

Comment:
Should this be a reset as well or is it a special case because it's connectionless?

Reply:
This should be replace.